test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -0,0 +1,204 @@
+        at the same time set an alarm value so less that
+        the hook alarms out and server executes the fail_action
+        After this check if vnodes are marked offline.
+        In case of a single mom reporting vnodes, it should mark
+        all the vnodes and mom as offline.
+        """
+        self.create_multi_vnodes(1, True)
+        self.create_mom_hook()
+
+        j1 = Job(TEST_USER)
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=self.moms.values()[0].shortname)
+        j1.set_sleep_time(1000)
+        jid = self.server.submit(j1)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
+
+        self.server.expect(
+            NODE, {ATTR_NODE_state: 'offline'},
+            id=self.moms.values()[0].shortname)
+        if self.is_cray is True:
+            vlist = []
+            vnl = self.server.filter(
+                VNODE, {'resources_available.vntype': 'cray_compute'})
+            vlist = vnl["resources_available.vntype=cray_compute"]
+            # Loop through each compute vnode in the list and check if state =
+            # offline
+            for v1 in vlist:
+                # Check that the vnode is in offline state
+                self.server.expect(
+                    VNODE, {'state': 'offline'}, id=v1, max_attempts=3,
+                    interval=2)
+
+        else:
+            self.server.expect(
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[0]')
+            self.server.expect(
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[1]')
+
+    def test_multi_mom_hook_failure_affects_vnode(self):
+        """
+        Run an execjob_begin hook that sleeps for sometime,
+        at the same time set an alarm value so less that
+        the hook alarms out and server executes the fail_action
+        After this check if vnodes are marked offline.
+        In case of a multiple mom reporting same set of vnodes,
+        it should not mark the vnodes and mom as offline because
+        there are other moms active and reporting same vnodes.
+        """
+        self.create_multi_vnodes(2, False)
+        self.create_mom_hook()
+
+        # set one  natural node to have higher ncpus than the other one
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            ATTR_rescavail + '.ncpus': '256'},
+                            id=self.moms.values()[0].shortname)
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            ATTR_rescavail + '.ncpus': '1'},
+                            id=self.moms.values()[1].shortname)
+
+        j1 = Job(TEST_USER)
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=self.moms.values()[0].shortname)
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=self.moms.values()[1].shortname)

```

Comment:
Should max_attempts and interval be set on these 2 expect() too?

Reply:
Done